libraries/stdlib/samples/test/samples/collections/arrays.kt

```diff
@@ -64,6 +64,159 @@ class Arrays {
 
     class Transformations {
 
+        @Sample
+        fun associate() {
+            data class Person(val firstName: String, val lastName: String)
+
+            val scientists = arrayOf(Person("Grace", "Hopper"), Person("Jacob", "Bernoulli"), Person("Johann", "Bernoulli"))
```

Comment:
Provided that [`associate` doc page](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/associate.html) shows overloads for both collections and all arrays, I suggest leaving only one sample for all arrays: the one for primitive arrays `associateArrayOfPrimitives`.

Reply:
Sounds good, I'll replace the array samples for objects.